spec/models/report_spec.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Report, type: :model do
+  describe '#editable?' do
+    let(:user) { create(:user) }
+    let(:report) { create(:report, user:) }
+
+    context '自分の日報の場合' do
+      subject { report.editable?(user) }
+
+      it { is_expected.to eq true }
+    end
+
+    context '他人の日報の場合' do
+      subject { report.editable?(other_user) }
+
+      let(:other_user) { create(:other_user) }
+
+      it { is_expected.to eq false }
+    end
+  end
+
+  describe '#created_on' do
+    let(:user) { create(:user) }
+    let(:report) { create(:report, user:) }
+
+    it '日付を返すこと' do
+      travel_to('2023-08-20 17:32:46'.in_time_zone) do
+        expect(report.created_on).to eq '2023-08-20'.to_date
+      end
```

Comment:
このテストコードを読み解くには、

- travel_toでシステム日時が変わる
- `report.created_on` を呼んだタイミングで遅延評価で`let(:report)`が呼ばれ、travel_toで指定した日付がcreated_atに入る
- よって created_on は travel_to で指定した日時と同じ日付になる

という流れを理解しないといけません。

RSpec初心者にはちょっとハードルが高いので、[minitest版](https://github.com/fjordllc/fjord-books_app-2023/pull/7/files#diff-8a327ede5b93ff905ead5dc6d960dc90bad6ffefd0dd7cd768921f950f952ac7R16-R17)と同じテスト方法でもいいのでは？と思いました。

```suggestion
      expect(report.created_on).to eq Date.current
```

Reply:
そうですね。ただ自分はこれまでminitest版のレビューをする際に、「時間を固定しないとテストを実行する際に丁度日付が変わった場合、その時だけテストがFAILする懸念があるのでなるべく時間を固定しましょう」ってコメントをするようにしていて、自分のレビューを受けた生徒からすると矛盾が生じるかもしれません 😅 

折衷案として、 `created_on` を呼ぶ前に `created_at` に固定した時間を入れておくようにしていますがどうでしょうか？